packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -62,6 +71,7 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
             `spectrum-Button--${buttonVariant}`,
             {
               'spectrum-Button--quiet': isQuiet,
+              'spectrum-Button--iconOnly': hasIcon && !hasLabel,
```

Comment:
probably could just use `!hasLabel`
wouldn't make much sense to have a button with no label AND no icon